snippets/csharp/System/NullReferenceException/Overview/Chain1.cs

```diff
@@ -1,62 +1,71 @@
-﻿// <Snippet6>
-using System;
+﻿using System;
 
-public class Example
+namespace Chain1
 {
-   public static void Main()
-   {
-      var pages = new Pages();
-      if (!String.IsNullOrEmpty(pages.CurrentPage.Title)) {
-         String title = pages.CurrentPage.Title;
-         Console.WriteLine("Current title: '{0}'", title);
-      }
-   }
-}
+    // <Snippet6>
+    public class Chain1Example
+    {
+        public static void Main()
+        {
+            var pages = new Pages();
+            if (!string.IsNullOrEmpty(pages.CurrentPage.Title))
+            {
+                string title = pages.CurrentPage.Title;
+                Console.WriteLine("Current title: '{0}'", title);
+            }
+        }
+    }
 
-public class Pages
-{
-   Page[] page = new Page[10];
-   int ctr = 0;
+    public class Pages
+    {
+        readonly Page[] _page = new Page[10];
+        int _ctr = 0;
 
-   public Page CurrentPage
-   {
-      get { return page[ctr]; }
-      set {
-         // Move all the page objects down to accommodate the new one.
-         if (ctr > page.GetUpperBound(0)) {
-            for (int ndx = 1; ndx <= page.GetUpperBound(0); ndx++)
-               page[ndx - 1] = page[ndx];
-         }
-         page[ctr] = value;
-         if (ctr < page.GetUpperBound(0))
-            ctr++;
-      }
-   }
+        public Page CurrentPage
+        {
+            get { return _page[_ctr]; }
+            set
+            {
+                // Move all the page objects down to accommodate the new one.
+                if (_ctr > _page.GetUpperBound(0))
+                {
+                    for (int ndx = 1; ndx <= _page.GetUpperBound(0); ndx++)
+                        _page[ndx - 1] = _page[ndx];
+                }
+                _page[_ctr] = value;
+                if (_ctr < _page.GetUpperBound(0))
+                    _ctr++;
+            }
+        }
 
-   public Page PreviousPage
-   {
-      get {
-         if (ctr == 0) {
-            if (page[0] == null)
-               return null;
-            else
-               return page[0];
-         }
-         else {
-            ctr--;
-            return page[ctr + 1];
-         }
-      }
-   }
-}
+        public Page PreviousPage
+        {
+            get
+            {
+                if (_ctr == 0)
+                {
+                    if (_page[0] == null)
```

Comment:
(pure style)  We could consider pattern matching here:

```suggestion
                    if (_page[0] is null)
```